vgen/src/main/java/io/gapi/vgen/csharp/CSharpGapicCodeLanguageProvider.java

```diff
@@ -0,0 +1,34 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gapi.vgen.csharp;
+
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+
+import io.gapi.vgen.ApiConfig;
+
+public class CSharpGapicCodeLanguageProvider extends CSharpGapicLanguageProvider<Interface> {
+
+  private final CSharpInterfaceViewing viewing = new CSharpInterfaceViewing();
```

Comment:
Ok I think I have figured out a little bit more of the intent here - you are preserving the property of the config files where only one class is specified, and that fully determines all of the behavior. This unfortunately creates the need to have a distinct class for every config file, though. I think a more flexible approach would be for the config file to specify two classes: the language provider (limited to the existing ones before this PR) plus a "ProtoElementViewing" (or whatever it ends up being called, per my other comment) class, which are combined together for the desired behavior.